### PR TITLE
Initial repo setup

### PR DESCRIPTION
**Pull request by MehulKChaudhari**
Date: 4/11/2024

Developer Name: @MehulKChaudhari 

---
## Description

This PR will add basic repo setup for frontend of feature-flag, (React Typescript Vite)

### Documentation Updated?

- [x] Yes
- [ ] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by MehulKChaudhari** [comment]
Note: Tests setup will come in subsequent PR, It would have increased the number of files changes and the scope of this PR that's why keeping it separate.

File: index.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/rds-logo.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite + React + TS</title>
```
Comment (prakashchoudhary07) [inline index.html]: Please change the title?

File: pnpm-lock.yaml
Comment (prakashchoudhary07) [inline pnpm-lock.yaml:1]: Are we sure we are going to use pnpm?
why not npm?

**Review by prakashchoudhary07 — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (MehulKChaudhari) [reply index.html]: Oops, Changed. 

Thanks for noticing :)

File: pnpm-lock.yaml
Comment (MehulKChaudhari) [reply pnpm-lock.yaml:1]: Yes, Pnpm has better features than npm. 

Its fast, Efficient Disk Space Usage, Better Handling of Peer Dependencies etc

**Review by prakashchoudhary07 — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: CI
```
Comment (prakashchoudhary07) [inline .github/workflows/ci.yml:1]: Please add a time out

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
```
Comment (prakashchoudhary07) [inline .github/workflows/ci.yml:30]: Please use latest node version

**Review by prakashchoudhary07 — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  "scripts": {
+    "dev": "vite",
+    "build": "tsc -b && vite build",
+    "lint": "eslint .",
+    "lint:fix": "eslint . --fix",
+    "format": "prettier --check .",
+    "format:fix": "prettier --write .",
+    "preview": "vite preview",
+    "test": "vitest",
+    "test:watch": "vitest --watch"
```
Comment (prakashchoudhary07) [inline package.json:15]: Please add coverage report command

**Pull request merged**